src/app/api/v1/user/breaches/bulk-resolve/route.ts

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { authOptions } from "../../../../utils/auth";
+import { getServerSession } from "next-auth";
+import { logger } from "../../../../../functions/server/logging";
+import {
+  BreachBulkResolutionRequest,
+  Subscriber,
+} from "../../../../../(nextjs_migration)/(authenticated)/user/breaches/breaches.js";
+import { getBreaches } from "../../../../../functions/server/getBreaches";
+import { getAllEmailsAndBreaches } from "../../../../../../utils/breaches";
+import {
+  getSubscriberByEmail,
+  setBreachResolution,
+} from "../../../../../../db/tables/subscribers";
+import appConstants from "../../../../../../appConstants";
+
+export async function PUT(req: NextRequest): Promise<NextResponse> {
+  const session = await getServerSession(authOptions);
+  if (!session?.user?.subscriber) {
+    return new NextResponse(
+      JSON.stringify({ success: false, message: "Unauthenticated" }),
+      { status: 401 },
+    );
+  }
+
+  if (typeof session?.user?.email === "string") {
```

Comment:
Nit, but the `?` is not really needed to `session.user` anymore (because it's guarded above and there's a `return`, Typescript compiler should be able to figure that one out).

But I am curious about why `email` not being a string gets a different HTTP response, should this check just be moved up and return a `401`?  Do we expect this to happen? Seems like something we might want to log.

Reply:
Oh yea! Good catch, need to get rid of this copy/paste